crates/router/src/core/payments/routing.rs

```diff
@@ -1295,7 +1295,11 @@ pub async fn perform_success_based_routing(
         .attach_printable("unable to fetch success_rate based dynamic routing configs")?;
 
         let success_based_routing_config_params = success_based_routing_config_params_interpolator
-            .get_string_val(success_based_routing_configs.params.as_ref());
+            .get_string_val(
+                success_based_routing_configs
+                    .params.as_ref()
+                    .ok_or(errors::RoutingError::SuccessBasedRoutingParamsNotFoundError)
+                    .attach_printable("params not found in success based routing config")?);
```

Comment:
Attach printable is redundant here? You are logging the same message that is present in error variants display impl